pkgs/applications/networking/bee/bee-clef.nix

```diff
@@ -0,0 +1,71 @@
+{ version ? "release", stdenv, lib, substituteAll
+, fetchFromGitHub, go-ethereum }:
+
+let
+
+  versionSpec = {
+    release = rec {
+      name = "bee-clef";
+      version = "0.4.4";
+      rev = "refs/tags/v${version}";
+      sha256 = "18fab0nxxsglrdykhb02m6vd7pgd6z4pqgfp5nmpxjcz7s8rfnml";
+    };
+  }.${version};
+
+in
+
+stdenv.mkDerivation {
+  pname = versionSpec.name;
+  version = versionSpec.version;
+
+  src = fetchFromGitHub {
+    owner = "ethersphere";
+    repo = "bee-clef";
+    rev = versionSpec.rev;
+    sha256 = versionSpec.sha256;
+  };
+
+  buildInputs = [ go-ethereum ];
+
+  patches = [
+    (substituteAll {
+      src = ./bee-clef.patch;
+      clef = "${go-ethereum}/bin/clef";
+      inherit outputDir;
+
+      # this fails, but i also need to somehow inject paths that are referring to files in $out 
+      #??? inherit out;
+    })
+  ];
+
+  buildPhase = " "; # we don't want the Makefile to be run
+
+  # this intentionally reproduces the same directory structure as it
+  # is in the upstream repo, but only copies the files that are
+  # needed.
+  installPhase = ''
+    mkdir -p $out/lib/systemd/system
+    cp packaging/clef-service $out
+    #TODO delme substituteAll ${./ensure-clef-account} $out/ensure-clef-account
+    cp ${./ensure-clef-account} $out/ensure-clef-account
```

Comment:
You don't need to distribute helper scripts for the nixos service in the package. Just use `writers.writeShellScript` in the service.

Reply:
well, it's an interesting situation here, because bee-clef is not a proper package. it's merely the files to run the `go-ethereum` package's `clef` executable as a service in a special configuration. the point of packaging up a `bee-clef` package form the upstream repo is to automatically inherit as much from them as possible.

would this new info justify the current setup for you?

Reply:
oh, another thing: that shell script contains a line that i couldn't properly escape inside the nix file. that's the reason i decided to put it in the package.

Reply:
Another possibility you could consider is installing the `bee-clef` files in the `go-ethereum` package as well, since they are mostly text files the impact on size should be minimal and it'd save a second package.

Reply:
go-ethereum is a completely different story. mixing in bee-clef stuff there wouldn't be the right thing.